unittests/test_cli.py

```diff
@@ -298,6 +298,27 @@ def test_check_sanity_failure(run_reframe, tmp_path):
     )
 
 
+def test_check_sanity_failure_dry_run_mode(run_reframe, tmp_path):
+    returncode, stdout, stderr = run_reframe(
+        checkpath=['unittests/resources/checks/frontend_checks.py'],
+        more_options=['-n', 'SanityFailureCheck', '-S', 'dry_run_mode=1']
+    )
+    assert 'PASSED' in stdout
+
+    # This is a normal failure, it should not raise any exception
+    assert 'Traceback' not in stdout
+    assert 'Traceback' not in stderr
+    assert returncode == 0
+    assert os.path.exists(
+        tmp_path / 'output' / 'generic' / 'default' /
+        'builtin' / 'SanityFailureCheck' / 'rfm_job.sh'
+    )
+    assert not os.path.exists(
+        tmp_path / 'output' / 'generic' / 'default' /
+        'builtin' / 'SanityFailureCheck' / 'rfm_job.out'
+    )
```

Comment:
I think that the creation of the output directory should skipped completely in the dry-run mode. And here we should check that the job script is generated in the stage directory.